packages/components/addon/components/hds/table/th-selectable.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { guidFor } from '@ember/object/internals';
+
+export default class HdsTableThSelectableComponent extends Component {
+  /**
+   * Generate a unique ID for the Checkbox
+   * @return {string}
+   */
+  checkboxId = 'checkbox-' + guidFor(this);
+
+  @action
+  onChange() {
+    console.log('onChange called!');
+
+    let { onChange } = this.args;
+
+    if (typeof onChange === 'function') {
+      onChange();
+    }
+  }
+
+  @action
+  didInsert(element) {
+    const dataScope = element.getAttribute('data-scope');
+
+    if (dataScope === 'row') {
```

Comment:
you don't need to pass this information down to the template (`.hbs`), store it as `data-` attribute, and then re-read it back: you already have this information here: try to console.log the value of `this.args.scope` in this function, and you will see what I mean